daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala

```diff
@@ -1570,6 +1539,104 @@ class EngineTest
         )
       }
     }
+
+    "unused disclosed contracts not saved to ledger" in {
```

Comment:
`LookupByKey` testing - needs to be reviewed

Reply:
if you do want to be exhaustive (the fact that you test lookupByKey & FetchByKey kind of suggests you do), then you also need exercise, exerciseByKey. On the other hand, I'm not sure we do need to be exhaustive. It seems like the tests for `inputContracts` should be exhaustive in making sure that that works for all of these and here we effectively just need one test that checks that the logic gets triggered and stuff gets filtered.

Reply:
My intention is not to be exhaustive (even though this is obviously the better practice to follow), I'm just testing the latter scenario (i.e. ..logic gets triggered and stuff gets filtered). My intent/goal is to provide sufficient confidence that unused disclosed contracts do not find themselves saved within the transaction.

If we were to be more exhaustive here, then absolutely all these other speedy commands would need to be tested.

As the current issue is about stopping garbage finding its way into the transaction, and we claim to resolve this problem by filtering over `inputContracts`, I think being exhaustive here would be overkill.